pmdarima/compat/sklearn.py

```diff
@@ -4,50 +4,40 @@
 #
 # Patch backend for sklearn
 
-from pkg_resources import parse_version
-
-import sklearn
-from sklearn.utils.validation import check_is_fitted
+from sklearn.exceptions import NotFittedError
 
 __all__ = [
-    'get_compatible_check_is_fitted',
+    'check_is_fitted',
     'safe_indexing',
 ]
 
 
-def get_compatible_check_is_fitted(estimator, attributes=None):
-    """Make the ``check_is_fitted`` function compatible across the
-    sklearn 0.21 to 0.22+ boundary.
+def check_is_fitted(estimator, attributes):
+    """Ensure the model has been fitted
+
+    Typically called at the beginning of an operation on a model that requires
+    having been fit. Raises a ``NotFittedError`` if the model has not been
+    fit.
 
+    This is an adaptation of scikit-learn's ``check_is_fitted``, which has been
+    changed recently in a way that is no longer compatible with our package.
 
     Parameters
     ----------
     estimator : estimator instance,
         The estimator that will be checked to see if it is fitted.
 
-    attributes : str, optional (default=None)
-        The attributes to be passed into the ``check_is_fitted`` function.
-
+    attributes : str or iterable
+        The attributes to check for
     """
-
-    # Use the version string from sklearn to determine how to use
-    # the `sklearn.utils.validation import check_is_fitted` function
-    # Determine if we should require a value for the attribute field
-    is_no_attribute_needed = False
-    if parse_version(sklearn.__version__) >= parse_version('0.22.0'):
-        is_no_attribute_needed = True
-        
-    # If we need to pass in `attributes` but it is not a str type raise error.
-    if not is_no_attribute_needed and not isinstance(attributes, str):
-        raise TypeError("Parameter `attributes` can only accept str type")
-
-    # If we don't need an attribute (i.e. sklearn.__version__ >= 0.22)
-    # We ignore `attributes` to bypass warning
-    if is_no_attribute_needed:
-        return check_is_fitted(estimator=estimator)
-
-    # Otherwise we use the attribute that was passed in regardless of version.
-    return check_is_fitted(estimator=estimator, attributes=attributes)
+    if isinstance(attributes, str):
+        attributes = [attributes]
+    if not hasattr(attributes, "__iter__"):
+        raise TypeError("attributes must be a string or iterable")
+    for attr in attributes:
+        if hasattr(estimator, attr):
+            return
```

Comment:
So just so I understand this, assuming this is passed an iterable of attributes, this returns after it encounters the first valid attribute? For the most part, it looks like we call this with strings, do we have an example with iterables?

Reply:
That's correct. I don't have an example, but I can see a situation (particularly in a transformer) where depending on the constructor hparams, we may set either one fit attribute or another, in which case we'd want to check that either exists.

Reply:
Cool, makes sense 👍